<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CAN Firmware Upgrade Tool</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/resources.qrc">:/icons/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBoxConnection">
      <property name="title">
       <string>CAN Connection</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="labelInterface">
         <property name="text">
          <string>Interface:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxInterface"/>
       </item>
       <item>
        <widget class="QLabel" name="labelChannel">
         <property name="text">
          <string>Channel:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxChannel"/>
       </item>
       <item>
        <widget class="QLabel" name="labelBaudRate">
         <property name="text">
          <string>Baud Rate (kbps):</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxBaudRate"/>
       </item>
       <item>
        <widget class="QPushButton" name="buttonRefresh">
         <property name="text">
          <string>Refresh</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonConnect">
         <property name="text">
          <string>Connect</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelConnectionStatus">
         <property name="text">
          <string>Disconnected</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxFirmware">
      <property name="title">
       <string>Firmware Upgrade</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="lineEditFirmwareFile">
           <property name="placeholderText">
            <string>Select firmware file...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="buttonBrowse">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBoxTestMode">
         <property name="text">
          <string>Test mode (will revert on next reboot)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonFlash">
         <property name="text">
          <string>Flash Firmware</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxBoard">
      <property name="title">
       <string>Board Commands</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="buttonGetVersion">
         <property name="text">
          <string>Get Version</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonReboot">
         <property name="text">
          <string>Reboot Board</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelVersion">
         <property name="text">
          <string>Version: N/A</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxLog">
      <property name="title">
       <string>Log</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QTextEdit" name="textEditLog">
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="html">
          <string notr="true"/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
